From 73969aa878c7d7ee08077e8e0f8997836c1ac112 Mon Sep 17 00:00:00 2001
From: Sherry Sun <sherry.sun@nxp.com>
Date: Mon, 11 Nov 2024 14:22:10 +0800
Subject: [PATCH 1/2] hostapd: Update defconfig based on v2.11 version

Update the hostapd defconfig file to enable some NXP WiFi specific
features based on v2.11 version.

Upstream-Status: Inappropriate [i.MX specific]

Signed-off-by: Sherry Sun <sherry.sun@nxp.com>
---
 hostapd/defconfig | 39 +++++++++++++++++++++------------------
 1 file changed, 21 insertions(+), 18 deletions(-)

diff --git a/hostapd/defconfig b/hostapd/defconfig
index 550db697b..4abe83854 100644
--- a/hostapd/defconfig
+++ b/hostapd/defconfig
@@ -48,7 +48,7 @@ CONFIG_LIBNL32=y
 CONFIG_RSN_PREAUTH=y
 
 # Support Operating Channel Validation
-#CONFIG_OCV=y
+CONFIG_OCV=y
 
 # Integrated EAP server
 CONFIG_EAP=y
@@ -115,9 +115,9 @@ CONFIG_EAP_TTLS=y
 #CONFIG_EAP_TEAP=y
 
 # Wi-Fi Protected Setup (WPS)
-#CONFIG_WPS=y
+CONFIG_WPS=y
 # Enable UPnP support for external WPS Registrars
-#CONFIG_WPS_UPNP=y
+CONFIG_WPS_UPNP=y
 # Enable WPS support with NFC config method
 #CONFIG_WPS_NFC=y
 
@@ -145,7 +145,7 @@ CONFIG_IPV6=y
 #CONFIG_RADIUS_TLS=y
 
 # IEEE Std 802.11r-2008 (Fast BSS Transition)
-#CONFIG_IEEE80211R=y
+CONFIG_IEEE80211R=y
 
 # Use the hostapd's IEEE 802.11 authentication (ACL), but without
 # the IEEE 802.11 Management capability (e.g., FreeBSD/net80211)
@@ -153,26 +153,26 @@ CONFIG_IPV6=y
 
 # Wireless Network Management (IEEE Std 802.11v-2011)
 # Note: This is experimental and not complete implementation.
-#CONFIG_WNM=y
+CONFIG_WNM=y
 
 # IEEE 802.11ac (Very High Throughput) support
-#CONFIG_IEEE80211AC=y
+CONFIG_IEEE80211AC=y
 
 # IEEE 802.11ax HE support
-#CONFIG_IEEE80211AX=y
+CONFIG_IEEE80211AX=y
 
 # IEEE 802.11be EHT support
 # CONFIG_IEEE80211AX is mandatory for setting CONFIG_IEEE80211BE.
 # Note: This is experimental and work in progress. The definitions are still
 # subject to change and this should not be expected to interoperate with the
 # final IEEE 802.11be version.
-#CONFIG_IEEE80211BE=y
+CONFIG_IEEE80211BE=y
 
 # Simultaneous Authentication of Equals (SAE), WPA3-Personal
-#CONFIG_SAE=y
+CONFIG_SAE=y
 
 # SAE Public Key, WPA3-Personal
-#CONFIG_SAE_PK=y
+CONFIG_SAE_PK=y
 
 # Remove debugging code that is printing out debug messages to stdout.
 # This can be used to reduce the size of the hostapd considerably if debugging
@@ -283,7 +283,7 @@ CONFIG_IPV6=y
 # internal = Internal TLSv1 implementation (experimental)
 # linux = Linux kernel AF_ALG and internal TLSv1 implementation (experimental)
 # none = Empty template
-#CONFIG_TLS=openssl
+CONFIG_TLS=openssl
 
 # TLS-based EAP methods require at least TLS v1.0. Newer version of TLS (v1.1)
 # can be enabled to get a stronger construction of messages when block ciphers
@@ -317,10 +317,10 @@ CONFIG_IPV6=y
 # Interworking (IEEE 802.11u)
 # This can be used to enable functionality to improve interworking with
 # external networks.
-#CONFIG_INTERWORKING=y
+CONFIG_INTERWORKING=y
 
 # Hotspot 2.0
-#CONFIG_HS20=y
+CONFIG_HS20=y
 
 # Enable SQLite database support in hlr_auc_gw, EAP-SIM DB, and eap_user_file
 #CONFIG_SQLITE=y
@@ -361,12 +361,12 @@ CONFIG_IPV6=y
 # For more details refer to:
 # https://wireless.wiki.kernel.org/en/users/documentation/acs
 #
-#CONFIG_ACS=y
+CONFIG_ACS=y
 
 # Multiband Operation support
 # These extensions facilitate efficient use of multiple frequency bands
 # available to the AP and the devices that may associate with it.
-#CONFIG_MBO=y
+CONFIG_MBO=y
 
 # Client Taxonomy
 # Has the AP retain the Probe Request and (Re)Association Request frames from
@@ -375,7 +375,7 @@ CONFIG_IPV6=y
 #CONFIG_TAXONOMY=y
 
 # Fast Initial Link Setup (FILS) (IEEE 802.11ai)
-#CONFIG_FILS=y
+CONFIG_FILS=y
 # FILS shared key authentication with PFS
 #CONFIG_FILS_SK_PFS=y
 
@@ -385,7 +385,7 @@ CONFIG_IPV6=y
 
 # Opportunistic Wireless Encryption (OWE)
 # Experimental implementation of draft-harkins-owe-07.txt
-#CONFIG_OWE=y
+CONFIG_OWE=y
 
 # Airtime policy support
 #CONFIG_AIRTIME_POLICY=y
@@ -415,6 +415,9 @@ CONFIG_IPV6=y
 # production use.
 #CONFIG_PASN=y
 
+CONFIG_SUITEB=y
+CONFIG_SUITEB192=y
+
 # Device Provisioning Protocol (DPP) (also known as Wi-Fi Easy Connect)
 CONFIG_DPP=y
 # DPP version 2 support
@@ -424,4 +427,4 @@ CONFIG_DPP2=y
 #CONFIG_DPP3=y
 
 # Wi-Fi Aware unsynchronized service discovery (NAN USD)
-#CONFIG_NAN_USD=y
+CONFIG_NAN_USD=y
-- 
2.37.1

